value.go

```diff
@@ -96,10 +96,18 @@ type Value interface {
 	// Truth returns the truth value of an object.
 	Truth() Bool
 
-	// Hash returns a function of x such that Equals(x, y) => Hash(x) == Hash(y).
+	// x.Hash(seed) returns a function of x such that
+	// Equals(x, y) => Hash(x, seed) == Hash(y, seed),
+	// where seed is an arbitrary value chosen by the caller.
+	//
 	// Hash may fail if the value's type is not hashable, or if the value
 	// contains a non-hashable value.
-	Hash() (uint32, error)
+	//
+	// A good implementation should incorporate seed into the result
```

Comment:
Why make this part of the interface? It would be simpler to have a mixing function inside `hashtable` and `hash` that combines the seed and the hash code. Is there a security or efficiency disadvantage to that?

Reply:
You can't do it using the old interface because it constrains the effective hash to be the composition of two functions that can be separately analyzed, h = hash(x) and mix(seed, h).  The new interface effectively selects from 2^32 different hash functions based on the context (the table itself. or the collection in which the value appears).